### ISSUE TITLE
Resource leak in CoalescingBufferQueue

### ISSUE THREAD

**Issue by normanmaurer** [netty/netty#3989]
@louiscryan @nmittler  could you check ?

```
14:36:10 16:36:10.252 [main] DEBUG io.netty.util.NetUtil - /proc/sys/net/core/somaxconn: 128
14:36:10 16:36:10.311 [nioEventLoopGroup-8-0] ERROR io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See http://netty.io/wiki/reference-counted-objects.html for more information.
14:36:10 Recent access records: 0
14:36:10 Created at:
14:36:10 	io.netty.buffer.CompositeByteBuf.<init>(CompositeByteBuf.java:63)
14:36:10 	io.netty.buffer.AbstractByteBufAllocator.compositeDirectBuffer(AbstractByteBufAllocator.java:191)
14:36:10 	io.netty.buffer.AbstractByteBufAllocator.compositeBuffer(AbstractByteBufAllocator.java:169)
14:36:10 	io.netty.channel.CoalescingBufferQueue.compose(CoalescingBufferQueue.java:156)
14:36:10 	io.netty.channel.CoalescingBufferQueue.remove(CoalescingBufferQueue.java:127)
14:36:10 	io.netty.channel.CoalescingBufferQueueTest.testWithVoidPromise(CoalescingBufferQueueTest.java:102)
14:36:10 	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
14:36:10 	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
14:36:10 	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:36:10 	java.lang.reflect.Method.invoke(Method.java:497)
14:36:10 	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
14:36:10 	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
14:36:10 	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
14:36:10 	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
14:36:10 	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
14:36:10 	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
14:36:10 	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
14:36:10 	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
14:36:10 	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
14:36:10 	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
14:36:10 	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
14:36:10 	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
14:36:10 	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
14:36:10 	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
14:36:10 	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
14:36:10 	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
14:36:10 	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
14:36:10 	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
```

**Comment by louiscryan** [comment]
Looking....

**Comment by louiscryan** [comment]
Test was a little profligate in that it did not release the composite buffers removed from the queue. This is a failing of the test and not the queue itself. Consumers of the queue are required to ensure that the buffers they take out of the queue are ultimately released.

**Comment by normanmaurer** [comment]
@louiscryan could you do a PR to fix the test :) ?

**Comment by louiscryan** [comment]
PR already created :)

**Comment by normanmaurer** [comment]
Fixed by f608a139cfdb0a62ba84d5b006c6351d92578083
